src/utils.ts

```diff
@@ -51,16 +56,18 @@ export async function fetchData<T>(url: string, body?: unknown): Promise<T> {
   }
 
   const resp = await fetch(url, options)
-  const json = await resp.json()
+  let json
 
-  if (!resp.ok) {
-    let errTxt = ''
-    try {
-      const err = json as ErrorResponse
-      errTxt = `${err.code}: ${err.message}`
-    } catch (e) {
-      errTxt = resp.statusText
+  try {
+    json = await resp.json()
+  } catch {
+    if (resp.headers && resp.headers.get('content-length') === '0') {
+      throw new Error('Invalid response content')
```

Comment:
Maybe add the status text?

```suggestion
      throw new Error('Empty response ' + resp.statusText)
```